catkin_ws/src/sim/src/unity_bridge.py

```diff
@@ -6,47 +6,45 @@
 from auv_msgs.msg import DeadReckonReport, UnityState
 from geometry_msgs.msg import Quaternion, Vector3
 from sbg_driver.msg import SbgImuData, SbgEkfQuat
-from std_msgs.msg import Float64
+from std_msgs.msg import Float64, Bool
 from tf import transformations
 import quaternion
 
 DEG_PER_RAD = (180 / np.pi)
 
 def cb_unity_state(msg):
-    pose_x = msg.position[0]
-    pose_y = msg.position[1]
-    pose_z = msg.position[2]
-    pose_quat_x = msg.orientation.x
-    pose_quat_y = msg.orientation.y
-    pose_quat_z = msg.orientation.z
-    pose_quat_w = msg.orientation.w
-
-    twist_linear_x = msg.velocity[0]
-    twist_linear_y = msg.velocity[1]
-    twist_linear_z = msg.velocity[2]
-    twist_angular_x = msg.angular_velocity[0]
-    twist_angular_y = msg.angular_velocity[1]
-    twist_angular_z = msg.angular_velocity[2]
+    pose_x = msg.position.z
+    pose_y = -msg.position.x
+    pose_z = msg.position.y
+    q_NED_imunominal_x = msg.orientation.x
+    q_NED_imunominal_y = msg.orientation.y
+    q_NED_imunominal_z = msg.orientation.z
+    q_NED_imunominal_w = msg.orientation.w
+
+    twist_angular_x = -msg.angular_velocity.z
+    twist_angular_y = msg.angular_velocity.x
+    twist_angular_z = -msg.angular_velocity.y
 
     isDVLActive = msg.isDVLActive
     isIMUActive = msg.isIMUActive
     isDepthSensorActive = msg.isDepthSensorActive
 
+
     # DVL - NWU
     if isDVLActive:
+        pub_dvl_sensor_status.publish(Bool(True))
         # Position
         position_NWU_auv = np.array([pose_x, pose_y, pose_z])
```

Comment:
Does the unity state give position in nwu, but rotation in ned? @frompotenza

Reply:
yeah its weird the way I had to send the info to get it to work. Position and twist are sent in unity's reference frame and converted directly to NWU with some negations and switching axes. The rotation we get is in NED. It's because the way unity converted reference frames was weird but this worked so I left it at that, not ideal but it works